rust/cache/src/cache.rs

```diff
@@ -13,13 +14,13 @@ use thiserror::Error;
 pub enum Cache<K, V>
 where
     K: Send + Sync + Clone + Hash + Eq + 'static,
-    V: Send + Sync + Clone + 'static,
+    V: Value,
```

Comment:
I think this `Value` trait should be part of the Cache trait? Also do we want to make it so that the cache can only be used with values that now their size? that may be hard for some types

Reply:
Discussed offline, we introduce a concept of weight and a value type that does not implement it takes on a default weight of 1